judo/utils/normalization.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (c) 2025 Robotics and AI Institute LLC. All rights reserved.
+
+from typing import TypedDict, Union
+
+import numpy as np
+
+
+class MinMaxNormalizerKwargs(TypedDict, total=False):
+    """Kwargs for MinMaxNormalizer."""
+
+    min: np.ndarray
+    max: np.ndarray
+    eps: float
+
+
+class RunningMeanStdNormalizerKwargs(TypedDict, total=False):
+    """Kwargs for RunningMeanStdNormalizer."""
+
+    init_std: float
+    min_std: float
+    max_std: float
+    eps: float
+
+
+class IdentityNormalizerKwargs(TypedDict, total=False):
+    """Kwargs for IdentityNormalizer."""
+
+    pass
+
+
+NormalizerKwargs = Union[IdentityNormalizerKwargs, MinMaxNormalizerKwargs, RunningMeanStdNormalizerKwargs]
+
+
+class Normalizer:
+    """Base class for normalizers."""
+
+    def __init__(self, dim: int) -> None:
+        """Initialize the normalizer.
+
+        Args:
+            dim: Dimension of the data.
+        """
+        self.dim = dim
+
+    def normalize(self, x: np.ndarray) -> np.ndarray:
+        """Normalize the data.
+
+        Args:
+            x: The data to normalize. Shape=(..., dim).
+        """
+        raise NotImplementedError
+
+    def denormalize(self, x: np.ndarray) -> np.ndarray:
+        """Denormalize the data.
+
+        Args:
+            x: The data to denormalize. Shape=(..., dim).
+        """
+        raise NotImplementedError
+
+    def update(self, x: np.ndarray) -> None:
+        """Update the normalizer.
+
+        Args:
+            x: The data to update the normalizer with. Shape=(batch_size, dim).
+        """
+        pass
+
+
+class IdentityNormalizer(Normalizer):
+    """Normalizer that does nothing."""
+
+    def __init__(self, dim: int) -> None:
+        """Initialize the normalizer."""
+        super().__init__(dim)
+
+    def normalize(self, x: np.ndarray) -> np.ndarray:
+        """Return the data as is."""
+        return x
+
+    def denormalize(self, x: np.ndarray) -> np.ndarray:
+        """Return the data as is."""
+        return x
+
+
+class MinMaxNormalizer(Normalizer):
+    """Normalizer that uses min and max values to scale the data to the range [-1, 1]."""
+
+    def __init__(self, dim: int, min: np.ndarray, max: np.ndarray, eps: float = 1e-8) -> None:
+        """Initialize the normalizer.
+
+        Args:
+            dim: Dimension of the data.
+            min: The minimum values of the data.
+            max: The maximum values of the data.
+            eps: Small value to prevent division by zero.
+        """
+        super().__init__(dim)
+        self.min = min
+        self.max = max
+        self.eps = eps
+
+        # Only normalize the dimensions that are not -inf or inf
+        self.norm_dims = np.where((self.min != -np.inf) & (self.max != np.inf))[0]
```

Comment:
we should check if there are excluded dimensions here and raise a warning if there are infs so the user knows to check their model description. you can also point them to use the running normalizer instead